viz/TraversabilityGridVisualization.cpp

```diff
@@ -0,0 +1,329 @@
+//
+// Copyright (c) 2015-2017, Deutsches Forschungszentrum für Künstliche Intelligenz GmbH.
+// Copyright (c) 2015-2017, University of Bremen
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+#include <vizkit3d/TraversabilityGridVisualization.hpp>
+#include <vizkit3d/ExtentsRectangle.hpp>
+#include <vizkit3d/ColorConversionHelper.hpp>
+
+#include <osg/Texture2D>
+
+
+using namespace ::vizkit3d;
+
+struct TraversabilityGridVisualization::Data
```

Comment:
I guess you added all the visualization functions to the Data struct since it was done in the MLS visualization equally.
The reason for this, in case of the MLS, was that it is templated in order to support all different kinds of MLS maps. This is not necessary here. Please move the functions to the TraversabilityGridVisualization class. You can use the Data struct to hold the TraversabilityGrid but it can as well be a member of TraversabilityGridVisualization.

Reply:
Ahh well. I left out DataHold since it wasn't needed without any templates but didn't question wether or not I actually needed Data in that case. Thanks!

Reply:
Right, here it is entirely unnecessary.